corehq/motech/repeaters/tasks.py

```diff
@@ -85,14 +116,16 @@ def check_repeaters():
             "commcare.repeaters.check.processing",
             timing_buckets=_check_repeaters_buckets,
         ):
-            for record in iterate_repeat_records(start):
-                if not _soft_assert(
-                    datetime.utcnow() < twentythree_hours_later,
-                    "I've been iterating repeat records for 23 hours. I quit!"
-                ):
+            # process repeat records by chunks until there are no more records to process
+            ids_for_partition = get_repeat_record_ids_for_partition(start, partition, total_partitions)
+            while len(ids_for_partition) > 0:
```

Comment:
This code would be a bit cleaner if `get_repeat_record_ids_for_partition` handled pagination, maybe using the `paginate_view` util that the old dbaccessor does. That way this can be a simple for loop with single call to the function, rather than a while loop, which I think is a bit harder to follow.

Reply:
Yup I agree with this. I think you can keep a lot of the old code that uses paginate_view and just make sure to pass in `include_docs=False` instead of `include_docs=True` (and expecting the resultant items to be `{'id': ...}` instead of a wrapped doc).

Ideally it would be written so that the original for loop is

```
for record in iterate_repeat_records_for_partition(start, partition, total_partitions):
    # all the code under here stays the same
```
and the contents of the for loop is left unchanged.

`iterate_repeat_records_for_partition` would then be basically

```
for ids_chunk in chunked(iterate_repeat_record_ids_for_partition(start, partition, total_partitions), 500):
    yield from iterate_repeat_records_for_ids(ids_chunk)
```

and `iterate_repeat_record_ids_for_partition` would be what you have above as `get_repeat_records_for_ids`, but returning a generator expression / `yield`ing records rather than returning a list.

Reply:
Thanks for the feedback. I understand why it's beneficial to use `paginate_view` for the initial fetch of doc ids. In terms of using `chunked` on the iteration of doc ids, is this so we can still take advantage of a bulk fetch of repeat records in `iterate_repeat_records_for_ids`?

Reply:
Yup. If you fetch docs one at a time you become i/o bound (the round trip for each http request becomes a dominant issue), and if you fetch too many docs at a time (for example, a million) then you become CPU and/or memory bound at the db level and at the application level as you have to read an enormous http payload into memory. So there's a sweet spot of fetching enough docs at a time to lower the overall contribution of network latency to the time it takes to fetch the data, while keeping each request payload a reasonable size. We've never quantified the whole spectrum to find the optimal number, but anecdotally 500 has worked well.

Reply:
Oh wow. Im surprised 500 is the number you would recommend. I would have suggested something in the 1-10k range. It is easy enough to test different parameters. It would be good to run a quick test and choose based on that.

Reply:
Oh yeah? @calellowitz you might have more experience guiding your intuition with this to be honest. I might have said pull 1-10k ids at a time (within `iterate_repeat_record_ids_for_partition` where it also filters by hash) and then chunk those to fetch only 500 or so docs at a time. But those are very much off the top of my head numbers, and if fetching 1000s of docs at a time has worked well for other uses then it would likely be fine here too.

Reply:
<img width="534" alt="Screen Shot 2020-10-15 at 12 30 07 PM" src="https://user-images.githubusercontent.com/15785053/96159063-36d10d80-0ee2-11eb-82a7-bf92a3aa4120.png">
It's only 1 iteration but this is using `paginate_view` with the chunk size specified and running through all of the records returns. This is with docs included, but saw similar behavior with docs not included just shorter times obviously)

Reply:
<img width="532" alt="Screen Shot 2020-10-15 at 12 20 42 PM" src="https://user-images.githubusercontent.com/15785053/96159335-857ea780-0ee2-11eb-9b20-0ab3e18cf4d5.png">
This is with docs **not** included

Reply:
10k seems reasonable for IDs, but with full docs it sounds like it might require a lot of memory (even if it's fast).  @calellowitz @dannyroberts do you have any rules of thumb for what constitutes acceptable memory use?